packages/commonwealth/server/util/verifySessionSignature.ts

```diff
@@ -1,346 +1,48 @@
-          isValid = false;
-        }
-      } else {
-        log.error(
-          `Address not matched. Generated ${generatedAddress}, found ${addressModel.address}.`,
-        );
-        isValid = false;
-      }
-    }
-  } else if (community.base === ChainBase.Ethereum) {
-    //
-    // ethereum address handling
-    //
-    try {
-      const signaturePattern = /^(.+)\/([A-Za-z0-9]+)\/(0x[A-Fa-f0-9]+)$/;
-      const signaturePatternMatch = signaturePattern.exec(signatureString);
-      if (signaturePatternMatch === null) {
-        throw new Error(
-          `Invalid signature: signature did not match ${signaturePattern}`,
-        );
-      }
-      const [, domain, nonce, signatureData] = signaturePatternMatch;
-
-      const siweMessage = createSiweMessage(
-        canvasSessionPayload,
-        domain,
-        nonce,
-      );
-
-      if (addressModel.block_info !== sessionBlockInfo) {
-        throw new Error(
-          `Eth verification failed for ${addressModel.address}: signed a different block than expected`,
-        );
-      }
-
-      const address = verifyMessage(siweMessage, signatureData);
-
-      isValid = addressModel.address.toLowerCase() === address.toLowerCase();
-      if (!isValid) {
-        log.info(
-          `Eth verification failed for ${addressModel.address}: does not match recovered address ${address}`,
-        );
-      }
-    } catch (e) {
-      log.info(
-        `Eth verification failed for ${addressModel.address}: ${e.stack}`,
-      );
-      isValid = false;
-    }
-  } else if (community.base === ChainBase.NEAR) {
-    //
-    // near address handling
-    //
-
-    // both in base64 encoding
-    const nacl = (await import('tweetnacl')).default;
-    const { signature: sigObj, publicKey } = JSON.parse(signatureString);
-
-    isValid = nacl.sign.detached.verify(
-      Buffer.from(sortedStringify(canvasSessionPayload)),
-      Buffer.from(sigObj, 'base64'),
-      Buffer.from(publicKey, 'base64'),
-    );
-  } else if (community.base === ChainBase.Solana) {
-    //
-    // solana address handling
-    //
-
-    // ensure address is base58 string length 32, cf @solana/web3 impl
-    try {
-      const decodedAddress = bs58.decode(addressModel.address);
-      if (decodedAddress.length === 32) {
-        const nacl = (await import('tweetnacl')).default;
-        isValid = nacl.sign.detached.verify(
-          Buffer.from(sortedStringify(canvasSessionPayload)),
-          bs58.decode(signatureString),
-          decodedAddress,
-        );
-      } else {
-        isValid = false;
-      }
-    } catch (e) {
-      isValid = false;
+  const signer = getSessionSignerForAddress(session.address);
+  let isValid = false;
+  try {
+    if (signer !== undefined) {
+      await signer.verifySession(CANVAS_TOPIC, session);
+      isValid = true;
     }
-  } else {
-    // invalid network
-    log.error(`invalid network: ${community.network}`);
```

Comment:
lets keep this log

Reply:
Not totally sure what you mean by adding it back here, but changing the current error logging to use CW logger